File: apps/backend/src/domain/entities/appointment.py
from dataclasses import dataclass, field
from datetime import datetime

from src.domain.value_objects.enums import AppointmentType, AppointmentStatus


@dataclass
class Appointment:
    id: str
    user_id: str
    patient_id: str
    date_time: datetime
    type: AppointmentType
    duration: int = 60
    status: AppointmentStatus = AppointmentStatus.SCHEDULED
    notes: str | None = None
    meeting_url: str | None = None
    created_at: datetime = field(default_factory=datetime.utcnow)
    updated_at: datetime = field(default_factory=datetime.utcnow)

File: packages/shared-types/src/appointment.ts
import type { AppointmentType, AppointmentStatus } from "./enums";

export interface Appointment {
  id: string;
  userId: string;
  patientId: string;
  dateTime: string;
  duration: number;
  type: AppointmentType;
  status: AppointmentStatus;
  notes: string | null;
  meetingUrl: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateAppointmentDto {
  patientId: string;
  dateTime: string;
  duration?: number;
  type: AppointmentType;
  notes?: string;
  meetingUrl?: string;
}

export interface UpdateAppointmentDto {
  dateTime?: string;
  duration?: number;
  type?: AppointmentType;
  status?: AppointmentStatus;
  notes?: string;
  meetingUrl?: string;
}
